server/jetstream_cluster.go

```diff
@@ -1457,18 +1457,22 @@ func (js *jetStream) clusterStreamConfig(accName, streamName string) (StreamConf
 }
 
 func (js *jetStream) metaSnapshot() []byte {
-	var streams []writeableStreamAssignment
-
 	js.mu.RLock()
 	cc := js.cluster
+	nsa := 0
+	for _, asa := range cc.streams {
+		nsa += len(asa)
+	}
+	streams := make([]writeableStreamAssignment, 0, nsa)
 	for _, asa := range cc.streams {
 		for _, sa := range asa {
 			wsa := writeableStreamAssignment{
-				Client:  sa.Client,
-				Created: sa.Created,
-				Config:  sa.Config,
-				Group:   sa.Group,
-				Sync:    sa.Sync,
+				Client:    sa.Client,
```

Comment:
I wonder if we need client here for snapshots? Both for streams and consumers?

Reply:
I think we do, because there are quite a few places that stream assignments check for the service account and other fields once rehydrated from the snapshot.

Reply:
You are correct we need account, but not much else I believe.